csi/controller.go

```diff
@@ -433,13 +439,17 @@ func (s *OsdCsiServer) DeleteVolume(
 		return nil, status.Error(codes.InvalidArgument, "Volume id must be provided")
 	}
 
-	err := s.driver.Delete(req.GetVolumeId())
-	if err != nil {
-		e := fmt.Sprintf("Unable to delete volume with id %s: %s",
-			req.GetVolumeId(),
-			err.Error())
-		dlog.Errorln(e)
-		return nil, status.Error(codes.Internal, e)
+	// If the volume is not found, then we can return OK
+	volumes, err := s.driver.Inspect([]string{req.GetVolumeId()})
```

Comment:
what if it is a not a "not found" error and something else?

Reply:
good point. I'll take a look to see if Inspect returns a known error code. If not, we may not be able to determine and may have to guess.

Reply:
It doesn't look as if kvdb will guarantee the value of kvdb.ErrNotFound for all backends. Instead the GetVal implementation seems to return whatever error is returned from the interface implementation. I only see the memory kvdb backend use kvdb.ErrNotFound.


cc @adityadani